commonmark-ext-image-attributes/src/test/java/org/commonmark/ext/image/attributes/ImageAttributesTest.java

```diff
@@ -0,0 +1,112 @@
+package org.commonmark.ext.image.attributes;
+
+import org.commonmark.Extension;
+import org.commonmark.parser.Parser;
+import org.commonmark.renderer.html.HtmlRenderer;
+import org.commonmark.testutil.RenderingTestCase;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.Set;
+
+public class ImageAttributesTest extends RenderingTestCase {
+
+    private static final Set<Extension> EXTENSIONS = Collections.singleton(ImageAttributesExtension.create());
+    private static final Parser PARSER = Parser.builder().extensions(EXTENSIONS).build();
+    private static final HtmlRenderer RENDERER = HtmlRenderer.builder().extensions(EXTENSIONS).build();
+
+    @Test
+    public void baseCase() {
+        assertRendering("![text](/url.png){height=5}",
+                "<p><img src=\"/url.png\" alt=\"text\" height=\"5\" /></p>\n");
+
+        assertRendering("![text](/url.png){height=5 width=6}",
+                "<p><img src=\"/url.png\" alt=\"text\" height=\"5\" width=\"6\" /></p>\n");
+
+        assertRendering("![text](/url.png){height=99px   width=100px}",
+                "<p><img src=\"/url.png\" alt=\"text\" height=\"99px\" width=\"100px\" /></p>\n");
+
+        assertRendering("![text](/url.png){width=100 height=100}",
+                "<p><img src=\"/url.png\" alt=\"text\" width=\"100\" height=\"100\" /></p>\n");
+
+        assertRendering("![text](/url.png){height=4.8 width=3.14}",
+                "<p><img src=\"/url.png\" alt=\"text\" height=\"4.8\" width=\"3.14\" /></p>\n");
+
+        assertRendering("![text](/url.png){Width=18 HeIgHt=1001}",
+                "<p><img src=\"/url.png\" alt=\"text\" Width=\"18\" HeIgHt=\"1001\" /></p>\n");
+
+        assertRendering("![text](/url.png){height=green width=blue}",
+                "<p><img src=\"/url.png\" alt=\"text\" height=\"green\" width=\"blue\" /></p>\n");
+    }
+
+    @Test
+    public void doubleDelimiters() {
+        assertRendering("![text](/url.png){{height=5}}",
+                "<p><img src=\"/url.png\" alt=\"text\" height=\"5\" /></p>\n");
+    }
+
+    @Test
+    public void mismatchingDelimitersAreIgnored() {
+        assertRendering("![text](/url.png){", "<p><img src=\"/url.png\" alt=\"text\" />{</p>\n");
+    }
+
+    @Test
+    public void unsupportedStyleNamesAreLeftUnchanged() {
+        assertRendering("![text](/url.png){j=502 K=101 img=2 url=5}",
+                "<p><img src=\"/url.png\" alt=\"text\" />{j=502 K=101 img=2 url=5}</p>\n");
+        assertRendering("![foo](/url.png){height=3 invalid}\n",
+                "<p><img src=\"/url.png\" alt=\"foo\" />{height=3 invalid}</p>\n");
+        assertRendering("![foo](/url.png){height=3 *test*}\n",
+                "<p><img src=\"/url.png\" alt=\"foo\" />{height=3 <em>test</em>}</p>\n");
+    }
+
```

Comment:
This test fails with an `ArrayIndexOutOfBoundsException` when rendering. I think the delimiter processor should ensure each key has a value.

```suggestion

    @Test
    public void styleWithNoValueIsIgnored() {
        assertRendering("![text](/url.png){height}",
                "<p><img src=\"/url.png\" alt=\"text\" />{height}</p>\n");
    }
        
```

Reply:
I've added a check on `attribute.length` to `ImageAttributesAttributeProvider` and
  `ImageAttributesDelimiterProcessor`, and also included your new test.

Thanks for pointing that out!